### DISCUSSION TITLE
how to export and save aggregated images

### DISCUSSION THREAD

**Discussion by Mark-Ste** [BodenmillerGroup/steinbock#245] [Q&A]
Hi!
For cell segmentation with DeepCell steinbock is aggregating the nuclear and cytoplasmic marker channels. Is there a way to export these aggregations e.g. as a tiff file that includes one nuclear and one cytoplasmic channel with the aggregated data?
Thanks!

**Comment by Milad4849** [comment]
Hi @Mark-Ste 
Steinbock currently does not do that. Using the package tifffile and the multi-channel tiff output of steinbock, you can easily generate the dual-channel tiffs that you want, something like this:

```
import tifffile
import numpy as np

multi_channel_image = tifffile.imread(path/to/tiff)

channel_m = multi_channel_image[m,:,:]  
channel_n = multi_channel_image[n,:,:] 

new_multi_channel_image = np.stack((channel_m, channel_n), axis=0)
tifffile.imwrite('new_multi_channel_image.tif', dual_channel_image)
```

The above assumes you have a single nuclear and a single cytoplasmic channels. If you want to take into account multiple channels for each, you would have to aggregate them before stacking them together.

**Reply by Mark-Ste** [reply]
Hi @Milad4849
Thanks for your help and the code, that makes it a lot easier for me. What package do you prefer for channel aggregation?
Thanks again!
